ext/exif/exif.c

```diff
@@ -215,6 +215,26 @@ zend_module_entry exif_module_entry = {
 ZEND_GET_MODULE(exif)
 #endif
 
+/* php_stream_read() may return early without reading all data, depending on the chunk size
+ * and whether it's a URL stream or not. This helper keeps reading until the requested amount
+ * is read or until there is no more data available to read. */
+/* FIXME: perhaps this should be moved to some more general place? */
```

Comment:
It would have to be changed to handle nonblocking and also improve blocking side to handle repeated delay as it could exceed the total timeout in the current form.